src/pim-types.ts

```diff
@@ -255,6 +264,57 @@ export class TaskType extends EventType {
     ret.color = this.color;
     return ret;
   }
+
+  public getNextOccurence(): TaskType | null {
+    if (!this.isRecurring()) {
+      return null;
+    }
+
+    const rrule = this.rrule.clone();
+
+    if (rrule.count && rrule.count <= 1) {
+      return null; // end of reccurence
+    }
+
+    rrule.count = null; // clear count so we can iterate as many times as needed
+    const recur = rrule.iterator(this.startDate ?? this.dueDate);
+    let next = recur.next();
+    while ((next = recur.next())) {
+      if (next.compare(ICAL.Time.now()) > 0) {
+        break;
+      }
+    }
+
+    if (!next) {
+      return null; // end of reccurence
+    }
+
+    const nextStartDate = this.startDate && next;
```

Comment:
I don't get this `&& next`... You already test that next is not null a few lines above, and why have this in the first place?

Reply:
I'm using short-circuit evaluation. `next` is a Time. What I am trying to do is assign `nextStartDate` to `next` only if the `startDate` exists. Same with `endDate`. This might not be clear from my code though

Reply:
It could be a ternary instead: `const nextStartDate = this.startDate ? next : undefined`

Reply:
Oh I forgot typescript does that. I think this code is not very obvious. Maybe your suggested alternative is better, yeah.